### PR TITLE
fix: concurrent reverts

### PR DESCRIPTION
**Pull request by gfyrag**

### CODE REVIEW

File: components/ledger/pkg/api/apierrors/errors.go [hunk 1]
```diff
@@ -64,7 +65,9 @@ func coreErrorToErrorCode(err error) (int, string, string) {
 	case
 		ledger.IsValidationError(err),
 		state.IsPastTransactionError(err),
-		runner.IsNoPostingsError(err):
+		runner.IsNoPostingsError(err),
+		errors.Is(err, revert.ErrAlreadyReverted),
```
Comment (paul-nicolas) [inline components/ledger/pkg/api/apierrors/errors.go:69]: In order to keep the same syntax, should we create function ? (not a big deal for me, but just asking !)

File: components/ledger/pkg/ledger/revert/reverter.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+type Reverter struct {
+	sync.Map
+	store       Store
+	runner      Runner
+	logIngester LogIngester
+}
+
+func (r *Reverter) RevertTransaction(ctx context.Context, id string) (*core.ExpandedTransaction, error) {
+	_, loaded := r.Map.LoadOrStore(id, struct{}{})
```
Comment (paul-nicolas) [inline components/ledger/pkg/ledger/revert/reverter.go:45]: We really have to be sure that there is one ledger instance !

File: components/ledger/pkg/api/apierrors/errors.go (hunk 1 above)
Comment (gfyrag) [reply components/ledger/pkg/api/apierrors/errors.go:69]: yes, agreed

File: components/ledger/pkg/ledger/revert/reverter.go (hunk 2 above)
Comment (gfyrag) [reply components/ledger/pkg/ledger/revert/reverter.go:45]: Sorry, I don't understand ^^

**Review by paul-nicolas — APPROVED** [review]

**Pull request merged**